### PR TITLE
Enable prometheus only on tenant & resource masters in scaleout arch

### PR DESCRIPTION
**Pull request by vinaykul**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
/kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**: This fixes a bug where admin cluster would not come up. It starts prometheus process on tenant and resource partition master nodes only.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: cluster/gce/util.sh [hunk 1]
```diff
@@ -1568,7 +1568,7 @@ EOF
     # Node-only env vars.
     cat >>$file <<EOF
 KUBERNETES_MASTER: $(yaml-quote "false")
-ENABLE_APISERVER_INSECURE_PORT: $(yaml-quote ${ENABLE_APISERVER_INSECURE_PORT:-true})
+ENABLE_APISERVER_INSECURE_PORT: $(yaml-quote ${ENABLE_APISERVER_INSECURE_PORT:-false})
```
Comment (yb01) [inline cluster/gce/util.sh:1571]: I hope we tested and ensure this default setting change doesn't carry any other impact.

**Review by yb01 — APPROVED** [review]
/lgtm

File: cluster/gce/util.sh (hunk 1 above)
Comment (vinaykul) [reply cluster/gce/util.sh:1571]: this is what it was before Peng's change.

**Pull request merged**